packages/agent_cli/src/main.rs

```diff
@@ -1,618 +1,699 @@
-        cmd = cmd.subcommand(Command::new("setup"));
-    }
-
-    cmd
-}
+use std::collections::{HashMap, HashSet};
+use std::error::Error;
+use std::fmt::{Display, Formatter};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4};
+use std::str::FromStr;
+use std::sync::Arc;
+use std::time::Duration;
+
+use clap::{arg, Command};
+use rand::Rng;
+use uuid::Uuid;
+
+use autorun::autorun;
+use playit_agent_core::api::api::*;
+use playit_agent_core::api::http_client::HttpClientError;
+use playit_agent_core::api::PlayitApi;
+use playit_agent_core::network::address_lookup::{AddressLookup, AddressValue};
+use playit_agent_core::tunnel::setup::SetupError;
+use playit_agent_core::tunnel_runner::TunnelRunner;
+use playit_agent_core::utils::now_milli;
+use playit_secret::PlayitSecret;
+
+use crate::match_ip::MatchIp;
+use crate::signal_handle::get_signal_handle;
+use crate::ui::{UI, UISettings};
+
+
+//imports (chroma)
+use std::process::Command as ProcessCommand;
+use std::str;
+use crossterm::style::Stylize;
+
+pub const API_BASE: &'static str = "https://api.playit.gg";
+
+pub mod util;
+pub mod autorun;
+pub mod playit_secret;
+pub mod match_ip;
+pub mod ui;
+pub mod signal_handle;
+
+
+
+
+
+//ping function !WINDOWS! (chroma)
+async fn ipcheck() -> Result<(), CliError> {
+
+    println!("Testing {}","ply.gg...".yellow());
+    let output = ProcessCommand::new("ping")
+        .arg("-n")
+        .arg("1")
+        .arg("ping.ply.gg")
+        .output();
+
+    match output {
+        Ok(output) if output.status.success() => {
+            println!("PLY.gg is {}","not blocked!".green());
+            check_ips("ip", &["na", "sa", "eu", "as", "in"]).await;
+        }
+        _ =>{
+            println!("PLY.gg might be {}","blocked!".red());
+            check_ips("209.25.140", &[""]).await;
+            check_ips("209.25.141", &[""]).await;
+            check_ips("209.25.142", &[""]).await;
+            check_ips("209.25.143", &[""]).await;
+            check_ips("23.133.216", &[""]).await;
+        }
+    }
+
+    Ok(())
+}
+
+async fn check_ips(base: &str, domains: &[&str]) {
+    for &domain in domains {
+        for i in 1..=255 {
+            let address = if domain.is_empty() {
+                format!("{}.{}", base, i)
+            } else {
+                format!("{}.ip.{}.ply.gg", i, domain)
+            };
+
+            let output = ProcessCommand::new("ping")
+                .arg("-n")
+                .arg("1")
+                .arg("ping.ply.gg")
+                .output();
+
+            if let Ok(output) = output {
+                if output.status.success() {
+                    println!("Ping to {} {}", address, "succeeded!".green());
+                } else {
+                    println!("Ping to {} was {}", address, "not successful!".red());
+                }
+            }
+        }
+    }
+}
+
+
+
+
+
+
+
+#[tokio::main]
+async fn main() -> Result<std::process::ExitCode, CliError> {

```

Comment:
really hard to the what's changed here due to how the diff is calculated. Hopefully moving the new code into a different file will clean it up.